Added index to be required as module

andfk opened the pull request:
Hi, first of all, thanks for this good job migrating some JS components to Angular. 

Im working with Webpack right now and noticed that when requiring the angular-materialize had some issues, so i solved them with these changes BUT they affects to the package.json config, so i will try to get both configs working together later. I hope to get some help from you with this as i dont know whats the finality of the src/index.js yet.
Will be great if this is published into NPM too for many others (like me) in the near future :)

Regards!

andfk committed:
Added index to be required as module

webbiesdk commented:
I suppose its to be used in an import like: 

`angular.module('whatEverApp', [require("angular-materialize")])`

In that case, it is properly good idea to add something in the readme about it.

webbiesdk commented:
Do you know if this is the standard way of exporting Angular-modules? 

And the finality of src/index.js is that it can always be changed, if I make something, or someone sends a pull-request. It is not final in any way.

andfk commented:
Well, in terms of exporting the module:

``` javascript
// there is the way of exporting directly the module as:
return angular.module('XXX') 
// and then
angular.module('anotherXXX', [moduleImported.name]);
// -------------------------------------------------
// or its name only 
return angular.module('XXX').name;
// and then
angular.module('anotherXXX', [moduleImported]); 
```

I like more the second approach of required/imported and used directly like in your example, of course :). I inspired in the implemenation from the angular-resource package.

webbiesdk commented:
Ok. 

If you could add a note in the README, stating how one would use this with imports, then I will merge it.

andfk commented:
Well, i did some changes and replaced the src/index.js with mine, maybe should i get a way to get both configs or maybe not needed?. Added some useful info in README + the webpack setup too. Let me know.

webbiesdk commented:
I think this is good, nicely done.

andfk commented:
Many thanks! :+1:

webbiesdk commented:
I just bumped the version to 0.1.2, and published it on npm.

andfk commented:
Cool! thanks
